parquet/src/main/scala/magnolify/parquet/ParquetField.scala

```diff
@@ -327,27 +350,33 @@ object ParquetField {
         }
       }
 
-      override protected val isGroup: Boolean = hasAvroArray
+      override protected def isGroup(conf: Configuration): Boolean =
+        groupAvroArrays || conf.getBoolean(
+          MagnolifyParquetProperties.WriteGroupedArrays,
+          MagnolifyParquetProperties.WriteGroupedArraysDefault
+        )
```

Comment:
I'm afraid this check in `conf` is very costly and has a significant impact on performance. memorization can help, but IMHO we should give the `configuration` when creating the `ParquetType` (same place we actually capture the deprecated `ParquetArray`), so we can compute this only once. WDYT ?